src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+	"Reward",
+	"Purchase",
+	"Shop",
+	"Loot",
+	"Combat"
+}
+local GAMEPASS_IDS = {} -- List of all gamepass ids in the game
+local CUSTOM_DIMENSIONS = {
+	-- Uncomment and fill each dimension as needed
+	-- Refer to https://docs.gameanalytics.com/advanced-tracking/custom-dimensions
+	-- for more information about dimensions
+	
+	-- DIMENSION_01 = {};
+	-- DIMENSION_02 = {};
+	-- DIMENSION_03 = {};
+}
+
+--[=[
+	@class AnalyticsService
+	@server
+	
+	Author: Javi M (dig1t)
+	
+	Description: Knit service that handles GameAnalytics API requests.
+	
+	Make sure to change the API keys before using this service.
+	Development keys should be used for testing and production keys should be used for release.
+	
+	Before using this service, make sure to configure the following variables inside AnalyticsService.lua:
+	- CURRENCIES: List of all in-game currencies
+	- BUILD: Game version
+	- GAME_KEY: GameAnalytics game key
+	- SECRET_KEY: GameAnalytics secret key
+	- LOG_DEV_PRODUCT_PURCHASES: Whether or not to automatically log developer product purchases
+	- RESOURCE_EVENT_TYPES: List of all resource event types (example: player gained coins in a mission is a
+	"Reward" event type, player purchasing coins with Robux is a "Purchase" event type)
+	- GAMEPASS_IDS: List of all gamepass ids in the game
+	- CUSTOM_DIMENSIONS: Custom dimensions to be used in GameAnalytics (refer to [GameAnalytics docs](https://docs.gameanalytics.com/advanced-tracking/custom-dimensions) about dimensions)
+	
+	Events that happen during a mission (kills, deaths, rewards) should be
+	tracked and logged after the event ends	to avoid hitting API limits.
+	For example, if a player kills an enemy during a mission, the kill should be
+	tracked and logged (sum of kills) at the end of the mission.
+	
+	Refer to [GameAnalytics docs](https://docs.gameanalytics.com/integrations/sdk/roblox/event-tracking) for more information and use cases.
+	
+	Using AnalyticsService to track events on the client:
+	```lua
+	-- Inside a LocalScript
+	local ReplicatedStorage = game:GetService("ReplicatedStorage")
+	local Players = game:GetService("Players")
+	
+	local Packages = ReplicatedStorage.Packages
+	local Knit = require(Packages.Knit)
+	
+	Knit.Start():await() -- Wait for Knit to start
+	
+	local AnalyticsService = Knit.GetService("AnalyticsService")
+	
+	local timesOpenedShop: number = 2
+	
+	Players.PlayerRemoving:Connect(function(player: Player)
+		if player == Players.LocalPlayer then
+			-- Before player leaves the game,
+			-- log the number of times the shop was opened
+			AnalyticsService.LogEvent:Fire({ -- Use AnalyticsService.LogEvent:Fire() to log a client-side event
+				event: "UIEvent:OpenedShop",
+				value: timesOpenedShop -- Number of times the player opened their shop
+			})
+		end
+	end)
+	```
+]=]
+local AnalyticsService = Knit.CreateService({
+	Name = "AnalyticsService";
+	Client = {
+		LogEvent = Knit.CreateSignal();
+	};
+})
+
+--[=[
+	@interface PlayerEvent
```

Comment:
Moonwave has a [type tag](https://eryn.io/moonwave/docs/TagList#type) - interface is for defining fields/accepted types of a table.